Framework/API/src/Algorithm.cpp

```diff
@@ -1622,6 +1628,26 @@ bool Algorithm::getAlgStartupLogging() const {
   return m_isAlgStartupLoggingEnabled;
 }
 
+void Algorithm::declareProperty(std::unique_ptr<Kernel::Property> p,
+                                const std::string &doc) {
+  if (isReserved(m_reservedList, p->name()))
```

Comment:
No, they are all created at the same time and only added to this reserved list in `declareIndexProperty`